android/BOINC/app/build.gradle

```diff
@@ -1,5 +1,45 @@
+plugins {
+    id 'org.ajoberstar.grgit' version '1.7.1'
+    id 'com.palantir.git-version' version '0.7.1'
+}
+
 apply plugin: 'com.android.application'
 
+import org.ajoberstar.grgit.Grgit
+
+// Use commit date as version code
+def buildVersionCode() {
+    def repo = Grgit.open()
+    def head = repo.head()
+
+    // Sanity check across git plugins
+    assert head.getAbbreviatedId(10) == versionDetails().gitHash : "Internal error: SHA1 mismatch!"
+
+    return head.time
```

Comment:
This is fine until 07/18/2036 where the maximum allowed value for `versionCode` (2,100,000,000) is reached. https://developer.android.com/studio/publish/versioning.html#appversioning